dataplane/strategy/strategy.go

```diff
@@ -54,13 +71,112 @@ func getStrategyTypeListImpl(client strategyTypesClient) []*model.StrategyType {
 	return resp.Payload
 }
 
-func getStrategyID(client strategyTypesClient, strategyNeeded string) string {
+func getStrategyTypeID(client strategyTypesClient, strategyNeeded string) string {
 	var strategyID string
 	for _, strategy := range getStrategyTypeListImpl(client) {
 		strategyType := swag.StringValue(strategy.Name)
 		if strategyType == strategyNeeded {
-			strategyID = strategy.ID.String()
+			return strategy.ID.String()
+		}
+	}
+	return strategyID
+}
+
+func getStrategyID(client strategyClient, name string) string {
```

Comment:
A getLoginStrategyID should be better name, because int this way we have two functions with the same name, but different behavior

Reply:
Its better to keep it startegyID.